packages/db/src/translate/executions.ts

```diff
@@ -3,10 +3,68 @@ import Knex from 'knex';
 import { RecordDoesNotExist } from '@cumulus/errors';
 import { ExecutionRecord } from '@cumulus/types/api/executions';
 import Logger from '@cumulus/logger';
-import { PostgresExecution } from '../types/execution';
+import { PostgresExecution, PostgresExecutionRecord } from '../types/execution';
 import { ExecutionPgModel } from '../models/execution';
 import { CollectionPgModel } from '../models/collection';
 import { AsyncOperationPgModel } from '../models/async_operation';
+const { removeNilProperties } = require('@cumulus/common/util');
+
+export const translatePostgresExecutionToApiExecution = async (
+  executionRecord: PostgresExecutionRecord,
+  knex: Knex,
+  collectionPgModel = new CollectionPgModel(),
+  asyncOperationPgModel = new AsyncOperationPgModel(),
+  executionPgModel = new ExecutionPgModel()
+): Promise<ExecutionRecord> => {
+  let parentArn: string | undefined;
+  let collectionId: string | undefined;
+  let asyncOperationId: string | undefined;
+
+  if (executionRecord.collection_cumulus_id) {
+    const collection = await collectionPgModel.get(knex, {
+      cumulus_id: executionRecord.collection_cumulus_id,
+    });
+    collectionId = `${collection.name}___${collection.version}`;
+  }
+  if (executionRecord.async_operation_cumulus_id) {
+    const asyncOperation = await asyncOperationPgModel.get(knex, {
+      cumulus_id: executionRecord.async_operation_cumulus_id,
+    });
+    asyncOperationId = asyncOperation.id;
+  }
+  if (executionRecord.parent_cumulus_id) {
+    const parentExecution = await executionPgModel.get(knex, {
+      cumulus_id: executionRecord.parent_cumulus_id,
+    });
+    parentArn = parentExecution.arn;
+  }
+
+  const name = executionRecord.arn.split(':').pop();
```

Comment:
Nit: Could we comment about what the intended result looks like or make this a helper method with an example?

Reply:
Oh or maybe just note that it's the `postfix`. That might be clear enough.

Reply:
It's probably not unreasonable to make this a helper, we're doing it *all over the place*.   I erred on the side of not adding it to @common given the idiom itself (if not the intent) is straightforward, but that's perhaps a mistake. 

In either case, this probably helps: edd75a016